Sources/SwiftMemcache/Extensions/ByteBuffer+IntegerAsASCII.swift

```diff
@@ -0,0 +1,37 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-memcache-gsoc open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-memcache-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-memcache-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+
+import NIOCore
+
+extension ByteBuffer {
+    /// Write `integer` into this `ByteBuffer` as ASCII digits, without leading zeros, moving the writer index forward appropriately.
+    ///
+    /// - parameters:
+    ///     - integer: The integer to serialize.
+    @inlinable
+    public mutating func writeIntegerAsASCII<T: FixedWidthInteger>(_ integer: T) {
+        let asciiZero = UInt8(ascii: "0")
+        var value = integer
+        var buffer: ContiguousArray<UInt8> = []
+        
+        repeat {
+            let digit = UInt8(value % 10)
+            buffer.insert(asciiZero + digit, at: 0)
+            value /= 10
+        } while value > 0
+
+        self.writeBytes(buffer)
```

Comment:
We definitely shouldn't do it like this. This allocates multiple times. I will follow up on Monday with a better approach.

Reply:
I see the insert is going allocate new memory each time, we could do better by calculating what the number of digits would be and then allocate the buffer once but then thats still an extra allocation. 

If we think about writing directly into the ByteBuffer w/o making any intermediary array. We can calculate the number of digits before hand, use `self.reserveCapacity` to reserve that space and then write the bytes directly into the ByteBuffer.

Reply:
Let's do the following. Let's actually use Int to String here and write a String for now and we create an issue that tracks us improving the perf of writing the datalen.

Reply:
Thinking about this more there are two approaches that should work:

1. We could write everything into a big tuple and then reverse write them into the byte buffer
2. We could write everything in reverse into the byte buffer and then just swap the bytes around.

I would say we should explore the second version since that should be quite easy to write. As a note Swift has the `quotientAndRemainder` method on Int that helps with doing the mod and division in one operation. You should also be able to make this code easier by not doing it on `FixedWithInteger` but rather just `Int`.

Reply:
For now I will use the Int to string / write a string and create an issue to track this so that I can be improved in a subsequent PR.